iodata/formats/fchk.py

```diff
@@ -435,3 +439,225 @@ def _triangle_to_dense(triangle: np.ndarray) -> np.ndarray:
         result[:irow + 1, irow] = triangle[begin:end]
         begin = end
     return result
+
+
+# The fchk file has a very rigid format, to dump the information are
+# theses functions, both scalars and arrays, integer and real(float) variables
+def _dump_integer_scalars(name: str, val: int, f: TextIO):
+    """Dumper for a scalar integer."""
+    print("{0:40}   I     {1:12d}".format(name, int(val)), file=f)
+
+
+def _dump_real_scalars(name: str, val: float, f: TextIO):
+    """Dumper for a scalar float."""
+    print("{0:40}   R     {1: 16.8E}".format(name, float(val)), file=f)
+
+
+def _dump_integer_arrays(name: str, val: np.ndarray, f: TextIO):
+    """Dumper for a array of integers."""
+    nval = val.size
+    if nval != 0:
+        np.reshape(val, nval)
+        print("{0:40}   I   N={1:12}".format(name, nval), file=f)
+        k = 0
+        for i in range(nval):
+            print("{0:12}".format(int(val[i])), file=f, end='')
+            k += 1
+            if k == 6 or i == nval - 1:
+                print("", file=f)
+                k = 0
+
+
+def _dump_real_arrays(name: str, val: np.ndarray, f: TextIO):
+    """Dumper for a array of float."""
+    nval = val.size
+    if nval != 0:
+        np.reshape(val, nval)
+        print("{0:40}   R   N={1:12}".format(name, nval), file=f)
+        k = 0
+
+        for i in range(nval):
+            print("{0: 16.8E}".format(val[i]), file=f, end='')
+            k += 1
+            if k == 5 or i == nval - 1:
+                print("", file=f)
+                k = 0
+
+
+@document_dump_one(
+    "Gaussian Formatted Checkpoint",
+    ['atcoords', 'atnums', 'energy', 'mo', 'obasis'],
+    ['atcharges', 'atcorenums', 'atfrozen', 'atgradient', 'athessian', 'atmasses', 'lot',
+     'one_rdms', 'obasis_name', 'extra', 'moments'])
```

Comment:
The required and optional `IOData` attributes need to be updated to reflect the behavior of the code. I believe it should be as follows, but please double check:
```suggestion
    ['atnums', 'atcorenums'],
    ['atcharges', 'atcoords', 'atfrozen', 'atgradient', 'athessian', 'atmasses', 
￼     'charge', 'energy', 'lot', 'mo', 'one_rdms', 'obasis', 'obasis_name',
     'extra', 'moments'])
```

Reply:
the changes are made